src/components/motor/Motor.ts

```diff
@@ -0,0 +1,17 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { motorApi } from '../../main'
+
+export interface Properties {
+  positionReporting: boolean
+}
+
+export interface Motor {
+  setPower(power: number, extra?:Map<string, any>):Promise<motorApi.SetPowerResponse>;
+  goFor(rpm:number, revolutions:number, extra?:Map<string, any>):Promise<motorApi.GoForResponse>;
+  goTo(rpm:number, positionRevolutions:number, extra?:Map<string, any>):Promise<motorApi.GoToResponse>;
+  resetZeroPosition(offset:number, extra?:Map<string, any>):Promise<motorApi.ResetZeroPositionResponse>;
+  motorStop(extra?:Map<string, any>) :Promise<motorApi.StopResponse>;
+  getProperties(extra?:Map<string, any>) :Promise<Properties>;
+  getPosition(extra?:Map<string, any>):Promise<number>;
+  isPowered(extra?:Map<string, any>):Promise<(number | boolean)[]>;
```

Comment:
```suggestion
  isPowered(extra?:Map<string, any>):Promise<[boolean, number]>;
```

Reply:
I tried doing Promise<[boolean, number]> but the type checker didnt like it. I think it has something to do with me returning an array of return values and the fact that technically the array could contain a number or a bool in either position.

Reply:
by golly you've done it